server/controllers/buyer/addFavoriteProduct.js

```diff
@@ -0,0 +1,11 @@
+const { addFavoriteProduct } = require('../../database/queries');
+
+const addFavoriteHandler = (req, res, next) => {
+  const { productId } = req.body;
+  const { userId } = req.params;
+  addFavoriteProduct(userId, productId)
+    .then(() => res.json({ status: 200, message: 'A new favorite product is added successfully' }))
+    .catch(next);
+};
```

Comment:
please, handle the case when the product exists, and send a message like 'the product already exists in your favorite list'